sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -140,7 +142,14 @@ class SQLContext(@transient val sparkContext: SparkContext)
 
   // TODO how to handle the temp function per user session?
   @transient
-  protected[sql] lazy val functionRegistry: FunctionRegistry = FunctionRegistry.builtin
+  protected[sql] lazy val functionRegistry: FunctionRegistry = {
```

Comment:
actually i think there is one more problem here: HiveContext doesn't load whatever is added here. Can you add a unit test to UDFSuite in Hive package, and fix this for Hive as well?